src/checkout/views/Billing/Page.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from "react";
+import { generatePath } from "react-router";
+
+import { FormAddressType } from "../../../components";
+import { UserContext } from "../../../components/User/context";
+import { findFormErrors, maybe } from "../../../core/utils";
+import {
+  CartSummary,
+  GuestAddressForm,
+  StepCheck,
+  Steps,
+  UserAddressSelector
+} from "../../components";
+import { CheckoutStep } from "../../context";
+import { paymentUrl } from "../../routes";
+import { CheckoutFormType } from "../../types";
+import { Checkout } from "../../types/Checkout";
+import { IBillingPageProps, IBillingPageState } from "./types";
+
+const computeMutationVariables = (
+  formData: FormAddressType,
+  checkout: Checkout,
+  shippingAsBilling: boolean
+) => {
+  const { shippingAddress } = checkout;
+  const data = shippingAsBilling
+    ? (shippingAddress as FormAddressType)
+    : formData;
+
+  return {
+    variables: {
+      billingAddress: {
+        city: data.city,
+        country: maybe(() => data.country.value, data.country.code),
+        countryArea: data.countryArea,
+        firstName: data.firstName,
+        lastName: data.lastName,
+        postalCode: data.postalCode,
+        streetAddress1: data.streetAddress1,
+        streetAddress2: data.streetAddress2
+      },
+      checkoutId: checkout.id
+    }
+  };
+};
+
+class View extends React.Component<IBillingPageProps, IBillingPageState> {
+  readonly state = {
+    checkout: null,
+    errors: [],
+    loading: false
+  };
```

Comment:
Consider switching to hooks

Reply:
i'd love to, but at first we need to bump up version of react